### ISSUE TITLE
Multiply operator doesn't entirely override arrays

### ISSUE THREAD

**Issue by TsvetanKonstantinov** [mikefarah/yq#725]
**Describe the bug**

Merge of arrays via the multiply operator (without using append) doesn't override all values of the array.

Version of yq: 4.5.1
Operating system: mac
Installed via: homebrew

**Input Yaml**

a.yml:
```yaml
a:
  - orig1
  - orig2
  - orig3
  - orig4
  - orig5
  - orig6
```

b.yml:
```yaml
a:
  - new1
  - new2
  - new3
```

**Command**
The command you ran:
```
yq eval-all -P 'select(fileIndex == 0) * select(fileIndex == 1)' a.yaml b.yaml
```

**Actual behavior**
```yaml
a:
  - new1
  - new2
  - new3
  - orig4
  - orig5
  - orig6
```

**Expected behavior**

```yaml
a:
  - new1
  - new2
  - new3
```

**Additional context**
The same issue was discussed for yq version 3 here: https://github.com/mikefarah/yq/issues/455
and it was fixed.

**Comment by mikefarah** [comment]
Yeah good point - it will need to a a modifier to `*`,  like how there's `?` and `+`. 

Maybe `o` so it would be something like:

```
yq eval-all -P 'select(fileIndex == 0) *o select(fileIndex == 1)' a.yaml b.yaml
```

?

**Comment by TsvetanKonstantinov** [comment]
That would work for me. However, I think that this should be the default behavior.
The current behavior is not intuitive, and it seems to me that it is only useful in some very specific use-cases.

**Comment by mikefarah** [comment]
Yeah looking at what `jq` that's it's default behavior too - happy to make that change.

**Comment by mikefarah** [comment]
Fixed in 4.6.0
